packages/adapters/txservice/src/aggregator.ts

```diff
@@ -678,25 +679,71 @@ export class RpcProviderAggregator {
-      try {
-        return await method(provider);
-      } catch (e: unknown) {
-        // TODO: With the addition of SyncProvider, this parse call may be entirely redundant. Won't add any compute,
-        // however, as it will return instantly if the error is already a NxtpError.
-        const error = parseError(e);
-        if (error.type === ServerError.type || error.type === RpcError.type || error.type === StallTimeout.type) {
-          // If the method threw a StallTimeout, RpcError, or ServerError, that indicates a problem with the provider and not
-          // the call - so we'll retry the call with a different provider (if available).
-          errors.push(error);
-        } else {
-          // e.g. a TransactionReverted, TransactionReplaced, etc.
-          throw error;
+    const handleError = (e: unknown) => {
+      // TODO: With the addition of SyncProvider, this parse call may be entirely redundant. Won't add any compute,
+      // however, as it will return instantly if the error is already a NxtpError.
+      const error = parseError(e);
+      if (error.type === ServerError.type || error.type === RpcError.type || error.type === StallTimeout.type) {
+        // If the method threw a StallTimeout, RpcError, or ServerError, that indicates a problem with the provider and not
+        // the call - so we'll retry the call with a different provider (if available).
+        errors.push(error);
+      } else {
+        // e.g. a TransactionReverted, TransactionReplaced, etc.
+        throw error;
+      }
+    };
+
+    const quorum = this.config.quorum ?? 1;
+    if (quorum > 1) {
+      // Consult ALL providers.
+      const results: (T | undefined)[] = await Promise.all(
+        this.providers.map(async (provider) => {
+          try {
+            return await method(provider);
+          } catch (e: unknown) {
+            handleError(e);
+            return undefined;
+          }
+        }),
+      );
+      // Filter out undefined results.
+      // NOTE: If there aren't any defined results, we'll proceed out of this code block and throw the
+      // RpcError at the end of this method.
+      const filteredResults: T[] = results.filter((item) => item !== undefined) as T[];
+      if (filteredResults.length > 0) {
+        // Pick the most common answer.
+        let counts: Map<T, number> = new Map();
+        counts = filteredResults.reduce((counts, item) => {
+          counts.set(item, (counts.get(item) ?? 0) + 1);
+          return counts;
+        }, counts);
+        const maxCount = Math.max(...Array.from(counts.values()));
+        if (maxCount < quorum) {
+          // Quorum is not met: we should toss this response as it could be unreliable.
+          throw new QuorumNotMet(maxCount, quorum, {
+            errors,
+            providersCount: this.providers.length,
+            responsesCount: filteredResults.length,
+          });
+        }
+        // Technically it could be multiple top responses...
```

Comment:
we should log if there are multiple conflicting top responses at least

Reply:
Sure, agreed :)